### PR TITLE
[ENG-3436] chore: Make some code refactoring for RBF in the extension

### PR DESCRIPTION
**Pull request by dhriaznov**
# 🔘 PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no API changes)

# 📜 Background
After the RBF UI & logic were added to the web-extension, we also developed it for the mobile app. And some things could be improved in its code like constants, typing, etc. (kudos to @fedeerbes for the mobile app PR review)

This PR is made to address those things for the web extension as well.

Issue Link: #[[ENG-3436](https://linear.app/xverseapp/issue/ENG-3436/make-some-code-refactoring-for-rbf-in-the-extension)]
Context Link (if applicable):

# 🔄 Changes
- Moved Speed Up tx custom fee component to a separate folder and created a separate file for its styles
- Removed the `initialFeeRate` property of the custom fee component that had the same value as `minimumRbfFee`
- Fixed typing for the Speed Up tx screen
- Moved the `handleKeyDownFeeRateInput` function to helpers to reuse it
- Added `EMPTY_LABEL` constant
- Used the `FiatAmountText` component instead of duplicating the `getFiatAmountString` function
- Moved the copy related to estimated RBF completion time to translations

Impact:
- This PR makes the RBF code more clean and robust, and shouldn't break any logic

# 🖼 Screenshot / 📹 Video
Include screenshots or a video demonstrating the changes. This is especially helpful for UI changes.

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: src/app/screens/speedUpTransaction/customFee/index.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    const response = await calculateTotalFee(feeRateInput);
+
+    if (response) {
+      setTotalFee(response.toString());
+    }
+  };
+
+  useEffect(() => {
+    fetchTotalFee();
+  }, [feeRateInput]);
```
Comment (teebszet) [inline src/app/screens/speedUpTransaction/customFee/index.tsx:64]:
```suggestion

  useEffect(() => {
    const fetchTotalFee = async () => {
      const response = await calculateTotalFee(feeRateInput);
  
      if (response) {
        setTotalFee(response.toString());
      }
    };
    fetchTotalFee();
  }, [feeRateInput, calculateTotalFee]);
```

I think if fetchTotalFee is not a function used anywhere but this useEffect, it should go inside to properly manage the trigger conditions

File: src/app/screens/speedUpTransaction/customFee/index.styled.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+export const FeeContainer = styled.div({
+  display: 'flex',
+  flexDirection: 'column',
+});
+
+export const ControlsContainer = styled.div`
+  display: flex;
+  gap: 12px;
+  margin: 24px 16px 40px;
+`;
```
Comment (teebszet) [inline src/app/screens/speedUpTransaction/customFee/index.styled.ts:81]: fyi: I have started moving some common layouts (defined in zeroheight) into common.styled.ts :D

File: src/app/screens/speedUpTransaction/customFee/index.styled.ts
Comment (teebszet) [inline src/app/screens/speedUpTransaction/customFee/index.styled.ts:1]: very clean, thank you!

File: src/app/screens/speedUpTransaction/customFee/index.tsx
Comment (teebszet) [inline src/app/screens/speedUpTransaction/customFee/index.tsx:1]: 👏 👏

File: src/app/screens/speedUpTransaction/index.tsx [hunk 3]
```diff
@@ -195,7 +206,7 @@ function SpeedUpTransactionScreen() {
     } catch (err: any) {
       console.error(err);
 
-      if (err?.response?.data && err?.response?.data.includes('insufficient fee')) {
+      if (err?.response?.data && err.response.data.includes('insufficient fee')) {
```
Comment (teebszet) [inline src/app/screens/speedUpTransaction/index.tsx:209]:
```suggestion
      if (err?.response?.data?.includes('insufficient fee')) {
```

yeah or this 😄

File: src/app/screens/speedUpTransaction/index.tsx [hunk 4]
```diff
@@ -128,7 +131,7 @@ function SpeedUpTransactionScreen() {
     } finally {
```
Comment (teebszet) [inline src/app/screens/speedUpTransaction/index.tsx:131]: I wonder if we actually want to start displaying some errors in toast for catch errors like these...

File: src/app/screens/speedUpTransaction/index.tsx [hunk 5]
```diff
@@ -381,55 +376,63 @@ function SpeedUpTransactionScreen() {
                     const priorityOrder = ['highest', 'higher', 'high', 'medium'];
                     return priorityOrder.indexOf(a[0]) - priorityOrder.indexOf(b[0]);
                   })
```
Comment (teebszet) [inline src/app/screens/speedUpTransaction/index.tsx:378]: maybe you could sort this before setRbfRecommendedFees() ?

**Review by teebszet — DISMISSED** [review]
thanks for taking the time to tidy the code! I had some more code style suggestions, but only suggestions.

and a couple of nits on (probably insignificant) react optimisations

File: src/app/screens/speedUpTransaction/customFee/index.tsx (hunk 1 above)
Comment (dhriaznov) [reply src/app/screens/speedUpTransaction/customFee/index.tsx:64]: When I apply the code you suggested, it starts lagging for some reason 🤔 let's probably keep the original version for now

File: src/app/screens/speedUpTransaction/index.tsx (hunk 4 above)
Comment (dhriaznov) [reply src/app/screens/speedUpTransaction/index.tsx:150]: makes sense! I added a todo comment for this

**Comment by dhriaznov** [comment]
Thanks for the review @teebszet! I updated the PR, and also included the fix from https://github.com/secretkeylabs/xverse-web-extension/pull/713 you created since it's related and would be better to test those 2 PRs together. Could you approve it again pls? 🙂

**Review by teebszet — APPROVED** [review]

**Comment by dhriaznov** [comment]
It would make sense to test and merge this PR together with https://github.com/secretkeylabs/xverse-web-extension/pull/727 in the next release (0.28.0)

**Comment by DuskaT021** [comment]
@dhriaznov this can be merged, tested per [agreement](https://secretkeylabs.slack.com/archives/C05QCJM9D1N/p1704799982533299?thread_ts=1704781971.329409&cid=C05QCJM9D1N)

**Pull request merged**